pulser/parametrized/variable.py

```diff
@@ -50,14 +49,14 @@ def __post_init__(self):
         self._clear()
 
     @property
-    def variables(self):
+    def variables(self) -> Dict:
         return {self.name: self}
 
-    def _clear(self):
+    def _clear(self) -> None:
         self.__dict__["value"] = None
         self.__dict__["_count"] += 1
 
-    def _assign(self, value):
+    def _assign(self, value: List) -> None:
```

Comment:
This one is actually tricky, it's 
```suggestion
    def _assign(self, value: Union[ArrayLike, str, float, int] ) -> None:
```
Import `ArrayLike` from `numpy.typing`